scipy/_lib/_array_api.py

```diff
@@ -392,7 +392,41 @@ def xp_unsupported_param_msg(param: Any) -> str:
 
 
 def is_complex(x: Array, xp: ModuleType) -> bool:
-    return xp.isdtype(x.dtype, 'complex floating')
+    """While xp.isdtype is not always available."""
+    try:
+        return xp.isdtype(x.dtype, 'complex floating')
+    except AttributeError:
+        return x.dtype in (xp.complex64, xp.complex128)
+
+
+def is_integer(x: Array, xp: ModuleType) -> bool:
+    """While xp.isdtype is not always available."""
+    return is_integer_dtype(x.dtype, xp)
+
+
+def is_integer_dtype(dtyp, xp: ModuleType) -> bool:
```

Comment:
likewise here, `xp` should always have `isdtype`, either because we have used `array_namespace`, or because we have used `np_compat` as the default.